lib/FIR/afforestation.cc

```diff
@@ -25,6 +25,10 @@
 
 namespace Fortran::FIR {
 namespace {
+template<typename A> Expression *ExprRef(const A &a) {
```

Comment:
You could consider adding an overload that takes `const common::Indirection<parser::Expr>`. That's pretty common the the parse tree, so it would simplify some of the places you call `ExprRef`.

Reply:
good idea